packages/core/src/application.ts

```diff
@@ -47,21 +47,22 @@ export class Application extends Context {
       }
     }
 
-    this._bindSequence();
+    this._bindSequence(options && options.sequence);
 
     this.handleHttp = (req: ServerRequest, res: ServerResponse) =>
       this._handleHttpRequest(req, res);
 
     this.bind('logError').to(this._logError.bind(this));
   }
 
-  protected _bindSequence(): void {
+  public sequence(sequence: Constructor<Sequence>) {
+    this._bindSequence(sequence);
+  }
+
+  protected _bindSequence(sequence?: Constructor<Sequence>): void {
     // TODO(bajtos, ritch, superkhau) figure out how to integrate this single
     // sequence with custom sequences contributed by components
-    const sequence = this.options && this.options.sequence ?
-      this.options.sequence :
-      Sequence;
-    this.bind('sequence').toClass(sequence);
+    this.bind('sequence').toClass(sequence || Sequence);
```

Comment:
I'm +1 for your proposal, it's more clear (I was trying to save code by reusing _bindSequence ;p).

As for your second proposal, does TS support default option values?

```
constructor(public options?={sequence: Sequence}: ApplicationOptions) {
```

The code above is probably wrong (not tested), but I'm guessing there is a way to do that so we don't have to do the `if (!options) options = {}` check either.

https://stackoverflow.com/questions/23314806/setting-default-value-for-typescript-object-passed-as-argument